scripts/build.sh

```diff
@@ -261,6 +261,9 @@ case "$product-$method-$platform" in
     ;;
 
   Firestore-xcodebuild-*)
+    # Make travis_retry more likely to succeed.
+    rm -rf ~/Library/Developer/Xcode/DerivedData
```

Comment:
This is a really steep price to pay. Are you certain this actually helps?

Note that we used to have this globally but it actually prevented Firestore from succeeding when retrying because rebuilding everything takes ~20 minutes on Travis.

FWIW, we've been tackling flakes in the Firestore build and found a bug in the emulator that was causing quite a few of them. Since we haven't exhausted this track I'd rather not do this nuclear thing--unless it's known to actually reduce flakiness somehow.